### ISSUE TITLE
Flsimulate 'value at 'map_file' is not of requested type' for running Map0 magnetic field and other custom fields

### ISSUE THREAD

**Issue by wausain** [SuperNEMO-DBD/Falaise#198]
With Falaise : 4.0.2, running simulations with the Map0 magnetic field which can be found in the simulation configuration or a custom field (.csv) leads to the following error.

_flsimulate : Setup/run of simulation threw exception
value at 'map_file' is not of requested type_

This worked perfectly fine with previous versions of Falaise and .csv files were supported. 

The settings I use in the simulation config are 

_"geometry:layout/if_basic/magnetic_field/is_active/type=Mapped" \
"geometry:layout/if_basic/magnetic_field/is_active/type/if_mapped/map=Map0" \ ..._

If this can be fixed quickly that would be great as I require simulations with the new external vertex generators for the magnetic fields meeting next friday.

Thanks

**Comment by drbenmorgan** [comment]
Could you post the full conf file you ran `flsimulate` with please? Could you also try running 4.0.2 using the script:

```
#@key_label  "name"
#@meta_label "type"
[name="flsimulate" type="flsimulate::section"]
numberOfEvents : integer = 100

[name="flsimulate.variantService" type="flsimulate::section"]
settings : string[2] = \
  "geometry:layout/if_basic/magnetic_field/is_active/type=Mapped" \
  "geometry:layout/if_basic/magnetic_field/is_active/type/if_mapped/map=Map0" 
```

please and post any results/full messages here? That works for me using 4.0.2 on macOS.

**Comment by drbenmorgan** [comment]
Oh, and which older versions were you running that worked?

**Comment by cherylepatrick** [comment]
I just tried your example config file and got the same error as Hamzah on * Falaise : 4.0.2 (90dd7f63-dirty) (MacOS 10.14.5)

**Comment by wausain** [comment]
I previously was using Falaise 4.0.0.

This is the sim config file I'm attempting to run: 

_#@format=datatools::configuration::variant
#@format.version=1.0
#@organization=snemo
#@application=falaise


[name="flsimulate" type="flsimulate::section"]
numberOfEvents : integer = 10



[name="flsimulate.variantService" type="flsimulate::section"]

settings : string[4] = \
"geometry:layout/if_basic/magnetic_field/is_active/type=Mapped" \
"geometry:layout/if_basic/magnetic_field/is_active/type/if_mapped/map=Map0" \
"vertexes:generator=pmt_xcalo_glass_bulk" \
"primary_events:generator=Tl208"_ 


After attempting to use the given sim conf i get the same error.

Also I'm attempting to run on CCLYON as I was doing before.

**Comment by drbenmorgan** [comment]
Thanks both, am able to reproduce on @cherylepatrick's version, and tip of `develop`, but _not_ releases 4.0.2/4.0.3. Tracking which commit broke it...

**Comment by drbenmorgan** [comment]
O.k., tracked down using [`git bisect`](https://git-scm.com/docs/git-bisect) to 42ed724b61737122fd161e2f2b180547827f07aa. Will prep a PR to reproduce and then fix the bug.

**Comment by cherylepatrick** [comment]
Thank you so much @drbenmorgan ! Do you know if there's anything else this might have affected that should also be tested, just in case? Or is it pretty well contained?

**Comment by drbenmorgan** [comment]
It looks well contained - it's related to the new configuration validation system. The config file, correctly, states that the `map_file` parameter is a path (`string as path`):

https://github.com/SuperNEMO-DBD/Falaise/blob/878d4ff0de9617937c652d80b7cfb5945e36105e/resources/snemo/demonstrator/geometry/GeometryPlugins/MagneticField/EMFields.conf#L15-L24

but the validation system wasn't updated to reflect that, expecting a bare string:

https://github.com/SuperNEMO-DBD/Falaise/blob/42ed724b61737122fd161e2f2b180547827f07aa/source/falaise/snemo/geometry/mapped_magnetic_field.cc#L353-L357

That won't propagate outside of `mapped_magnetic_field`. I'd thought that we had a test exercising use of the mapped field, but obviously not! At least it'll now be added.

**Comment by cherylepatrick** [comment]
Ah, great! Thanks for checking and I guess we will be ready for the future :)

**Comment by drbenmorgan** [comment]
Hi @wausain, this should now be fixed on the `develop` branch by c0ebba2. If you update your Falaise build and/or install to that, it should be good to go.
